ansible/roles/test/files/ptftests/vlan_test.py

```diff
@@ -0,0 +1,268 @@
+                                                                 dst_mac, self.dataplane.get_mac(0, target["port_index"]),
+                                                                 src_ip, target["peer_ip"], 63)
+                # Test for for indirectly-connected routing
+                src_ip = vlan_port["permit_vlanid"][str(vlan_id)]["remote_ip"]
+                for target in target_list:
+                    if vlan_id == target["vlan_id"]:
+                        # Skip same VLAN forwarding
+                        continue
+                    self.send_icmp_packet(vlan_port, vlan_id if vlan_id != vlan_port["pvid"] else 0,
+                                          src_mac, dst_mac, src_ip, target["remote_ip"])
+                    self.verify_icmp_packets_from_specified_port(target["port_index"], 
+                                                                 target["vlan_id"] if target["vlan_id"] != target["pvid"] else 0,
+                                                                 dst_mac, self.dataplane.get_mac(0, target["port_index"]),
+                                                                 src_ip, target["remote_ip"], 63)
+
+        # Test case #5
+        # Send ICMP packets to VLAN interfaces. 
+        # Verify ICMP reply packets can be received from ingress port.
+        self.log("Test case #5 starting ...")
+        for vlan_port in vlan_ports_list:
+            src_port = vlan_port["port_index"]
+            src_mac = self.dataplane.get_mac(0, src_port)
+            dst_mac = self.router_mac
+            for vlan_id in map(int, vlan_port["permit_vlanid"].keys()):
+                src_ip = vlan_port["permit_vlanid"][str(vlan_id)]["peer_ip"]
+                for vlan_intf in vlan_intf_list:
+                    if int(vlan_intf["vlan_id"]) != vlan_id:
+                        continue
+                    dst_ip = vlan_intf["ip"].split("/")[0]
+                    self.send_icmp_packet(vlan_port, vlan_id if vlan_id != vlan_port["pvid"] else 0,
+                                          src_mac, dst_mac, src_ip, dst_ip)
+                    exp_pkt = simple_icmp_packet(eth_src=self.router_mac,
+                                           eth_dst=src_mac,
+                                           dl_vlan_enable=True if vlan_id != vlan_port["pvid"] else False,
+                                           vlan_vid=vlan_id if vlan_id != vlan_port["pvid"] else 0,
+                                           vlan_pcp=0,
+                                           ip_dst=src_ip,
+                                           ip_src=dst_ip,
+                                           icmp_type=0,
+                                           icmp_code=0)
+
+                    masked_exp_pkt = Mask(exp_pkt)
+                    masked_exp_pkt.set_do_not_care_scapy(scapy.IP, "id")
+
+                    verify_packets(self, masked_exp_pkt, list(str(src_port)))
```

Comment:
can you print similiar log here?

self.log("Verify packet from ports ???")

I do not find the verify log in test 5.